sound/soc/sof/ops.h

```diff
@@ -312,6 +312,18 @@ snd_sof_pcm_platform_pointer(struct snd_sof_dev *sdev,
 		return 0;
 }
 
+static inline int
+snd_sof_dai_config(struct snd_sof_dev *sdev,
+			struct snd_soc_dai_link *link,
+			struct sof_ipc_dai_config *config,
+			int direction)
+{
+	 if (sdev->ops && sdev->ops->dai_config)
+		 return sdev->ops->dai_config(sdev, link, config, direction);
```

Comment:
I am not sure this is right? We will have multiple dais, some of which will need a config and others will not. The level of indirection should be at the dai level and probably part of the hw_params?

Reply:
We made a agreement that it should be done as SSP at config time.
My last PR is done at hw_params. But it doesn't work for tone because there maybe
no hw_params for tone.

Reply:
@ranj063 can you comment on the tone? I am pretty sure you did a modification to pass the hw_params.

Reply:
@ranj063  yes, I am also very curious about this. I got some info from keyon only

Reply:
@RanderWang @plbossart sorry I havent had the time to review this PR yet. But seems like the code is outdated now and I'm lost on what the original question was about.

Reply:
@plbossart @RanderWang yes, the tone component didnt work for the longest time because I wasnt setting hw_params. So in order to set the hw_params, we introduced the concept of a virtual PCM FE (a pcm that is not registered with ALSA) which would be used to set the hw_params.

Reply:
@ranj063 do you have chance to talk about this(virtual PCM) in alsa-devel mailing list? That's a good idea to handle kinds of problems like this tone one, it would be good if we can align this solution from Takashi and Mark before implement SOF specific coding.

Reply:
@keyonjie not yet. I'll have to re-visit the tone patches soon.